components/CovidMap.js

```diff
@@ -0,0 +1,702 @@
+                      thirdClusterer: thirdClusterer,  
+                      selectUpdate: false
+                    });
+    });
+  }
+
+  handleOptionChange(event) {
+    var selectedOption = event.target.value; 
+    if(selectedOption == 'all'){
+      var firstLegendText = "All Deaths";
+    } else if(selectedOption == 'facility') {
+      var firstLegendText = "County Facilities";
+      var secondLegendText = "State Facility";
+      var thirdLegendText = "Federal Facility";
+    } else if(selectedOption == 'age') {
+      var firstLegendText = "Ages Under 35";
+      var secondLegendText = "Ages 35-65";
+      var thirdLegendText = "Ages over 65";
+    } else if(selectedOption == 'ethnicity') {
+      var firstLegendText = "White Deaths";
+      var secondLegendText = "Black Deaths";
+      var thirdLegendText = "Hispanic Deaths";
+    }
+    this.setState({ selectedOption: selectedOption,
+                    selectUpdate: true,
+                    firstLegendText: firstLegendText,
+                    secondLegendText: secondLegendText,
+                    thirdLegendText: thirdLegendText
+                  });
+  }
+
+  render () {
+    const { map, firstClusterer, secondClusterer, thirdClusterer, fetchedMap, data, selectedOption, firstLegendText, secondLegendText, thirdLegendText } = this.state;
+    return (
+      <div id="map-container">
+        <div id="map" style={mapStyle} className="map"></div>
+        <div id="form" style={formStyle} className="form">
+        <form>
+          <div className="form-check">
+            <label>
+              <input type="radio" name="all" value="all" checked={this.state.selectedOption === "all"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              All Deaths
+            </label>
+          </div>
+          <div onChange={this.handleOptionChange.bind(this)} className="form-check">
+            <label>
+              <input type="radio" name="facility" value="facility" checked={this.state.selectedOption === "facility"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              By Facility
+            </label>
+          </div>
+          <div onChange={this.handleOptionChange.bind(this)} className="form-check">
+            <label>
+              <input type="radio" name="age" value="age" checked={this.state.selectedOption === "age"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              By Age
+            </label>
+          </div>
+          { 1 ? null :
+          <div onChange={this.handleOptionChange.bind(this)} className="form-check">
+            <label>
+              <input type="radio" name="ethnicity" value="ethnicity" checked={this.state.selectedOption === "ethnicity"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              By Ethnicity
+            </label>
+          </div>
+          }
```

Comment:
What do you think about dumping all of the code related to filtering by ethnicity if we don't plan to include this button?

Reply:
Yes, I think it would make things look slightly better. I also think this needs to be refactored to make adding additional filters easier in the future.